x-pack/plugins/profiling/server/routes/topn.ts

```diff
@@ -44,46 +45,41 @@ export async function topNElasticSearchQuery({
   const filter = createCommonFilter({ projectID, timeFrom, timeTo, kuery });
   const targetSampleSize = 20000; // minimum number of samples to get statistically sound results
 
-  const eventsIndex = await logExecutionLatency(
+  const eventsIndex = await findDownsampledIndex({
     logger,
-    'query to find downsampled index',
-    async () => {
-      return await findDownsampledIndex(logger, client, index, filter, targetSampleSize);
-    }
-  );
+    client,
+    index,
+    filter,
+    sampleSize: targetSampleSize,
+  });
 
-  const resEvents = await logExecutionLatency(
-    logger,
-    'query to fetch events from ' + eventsIndex.name,
-    async () => {
-      return await client.search(
-        {
-          index: eventsIndex.name,
-          size: 0,
-          query: filter,
-          aggs: {
-            histogram: autoHistogramSumCountOnGroupByField(searchField, n),
-          },
-        },
-        {
-          // Adrien and Dario found out this is a work-around for some bug in 8.1.
-          // It reduces the query time by avoiding unneeded searches.
-          querystring: {
-            pre_filter_shard_size: 1,
-          },
-        }
-      );
-    }
-  );
+  const resEvents = await client.search('get_top_n_histogram', {
+    index: eventsIndex.name,
+    size: 0,
+    query: filter,
+    aggs: {
+      histogram: autoHistogramSumCountOnGroupByField(searchField, n),
+    },
+    // Adrien and Dario found out this is a work-around for some bug in 8.1.
+    // It reduces the query time by avoiding unneeded searches.
+    pre_filter_shard_size: 1,
```

Comment:
Since we are past 8.1, does this still apply ?

Reply:
Do you have a link to the issue where it's fixed?

Reply:
So it looks like this came from when we did composite aggregations. See https://elastic.slack.com/archives/C035P7UMXR7/p1651064160824109?thread_ts=1650984764.314609&cid=C035P7UMXR7

The mentioned issue is still open and I wonder if we can avoid the cluster setting `search.max_buckets` without doing pagination (= back to composite aggs).

Reply:
Maybe it's good to leave it as it. We have to work on it (or rethink) before going public.